clearwater-infrastructure/usr/share/clearwater/infrastructure/scripts/hostname

```diff
@@ -37,13 +37,20 @@
 # Read our config file.
 . /etc/clearwater/config
 
+grep -v ' #+clearwater-infrastructure$' /etc/hosts > /tmp/hosts.$$
+
 # Check if the public hostname is actually an IPv4 address or an IPv6 address.
 if echo $public_hostname | egrep -v -q -e '^[0-9.]+$' -e '^[0-9A-Fa-f:]+$'
 then
   # If not, get the first (most-specific) part, and set that as the server's internal hostname.
-  echo $public_hostname | sed -e 's/\..*//g' >/etc/hostname
-  hostname $(cat /etc/hostname)
-  grep -v ' #+clearwater-infrastructure$' /etc/hosts >/tmp/hosts.$$
-  mv /tmp/hosts.$$ /etc/hosts
-  echo $local_ip $(cat /etc/hostname) '#+clearwater-infrastructure' >>/etc/hosts
+  echo $public_hostname | sed -e 's/\..*//g' > /etc/hostname
+
+  # Also, ensure that the public hostname is resolvable
+  echo $local_ip $public_hostname '#+clearwater-infrastructure' >> /tmp/hosts.$$
```

Comment:
OK, I've updated and retested:

- with a nonsensical and unresolvable hostname in public_hostname, sudo, sprout and Cassandra all continue to work
- with sensible hostnames, the live tests work
- we also only use the initial part of the hostname in the Bash command prompt, which is nice and preserves that part of the user experience

I'll merge this unless you have further questions.